vulnfeeds/cpp/main.go

```diff
@@ -32,6 +38,33 @@ func IsRepoURL(url string) bool {
 	return re.MatchString(url)
 }
 
+// Returns the base repository URL
+func Repo(u string) (string, bool) {
+	parsedURL, err := url.Parse(u)
+	if err != nil {
+		Logger.Fatalf("%v", err)
+	}
+	// GitHub and GitLab URLs are structured one way, e.g.
+	// https://github.com/MariaDB/server/commit/b1351c15946349f9daa7e5297fb2ac6f3139e4a8
+	// https://gitlab.freedesktop.org/virgl/virglrenderer/-/commit/b05bb61f454eeb8a85164c8a31510aeb9d79129c
+	// https://gitlab.com/qemu-project/qemu/-/commit/4367a20cc4
```

Comment:
I think there's also ones like `https://github.com/google/osv.dev/pull/738` in these sometimes. We should make this work too.

Reply:
Done.